Libraries/LibWeb/HTML/HTMLLIElement.h

```diff
@@ -20,7 +22,16 @@ class HTMLLIElement final : public HTMLElement {
     virtual ~HTMLLIElement() override;
 
     // https://www.w3.org/TR/html-aria/#el-li
-    virtual Optional<ARIA::Role> default_role() const override { return ARIA::Role::listitem; }
+    virtual Optional<ARIA::Role> default_role() const override
+    {
+        if (first_ancestor_of_type<HTML::HTMLUListElement>() || first_ancestor_of_type<HTML::HTMLOListElement>())
+            return ARIA::Role::listitem;
```

Comment:
I *think* the spec may be saying that we should check if any ancestor has a `list` role here.

If I open the accessibility tab of the Firefox debugger I can see that `#test` has a role of "listitem" here:
```html
<!DOCTYPE html>
<div role="listitem">
    <div role="list">
        <div id="test" role="listitem">
        </div>
    </div>
</div>
```

but a role of "generic" here:
```html
<!DOCTYPE html>
<div role="listitem">
    <div role="list">
        <div id="test" role="listitem">
        </div>
    </div>
</div>
```
This is really outside my area of expertise though, so I may be way off-base here.

Reply:
You’re right — it should be checking for the `list` role instead. Made it so ✅